src/api/spec/controllers/webui/package_controller_spec.rb

```diff
@@ -2,20 +2,89 @@
+
+      it "creates a submit request" do
+        expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: package.name)).to exist
+      end
+    end
+
+    context "having whitespaces in parameters" do
+      before do
+        login(user)
+        post :submit_request, { project: " #{source_project} ", package: " #{package} ", targetproject: " #{target_project} " }
+      end
+
+      it { expect(flash[:notice]).to match("Created .+submit request \\d.+to .+#{target_project}") }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: package)) }
+
+      it "creates a submit request" do
+        expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: package.name)).to exist
+      end
+    end
+
     context 'not successful' do
       before do
         login(user)
         post :submit_request, { project: source_project, package: source_package, targetproject: target_project.name }
       end
 
       it { expect(flash[:error]).to eq('Unable to submit: The source of package home:tom/my_package is broken') }
-      it { expect(BsRequestActionSubmit.where(target_project: target_project, target_package: source_package).count).to eq(0) }
+      it { expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: source_package.name)).not_to exist }
+    end
+
+    context "a submit request that fails due to validation errors" do
+      let(:unconfirmed_user) { create(:user) }
+
+      before do
+        login(unconfirmed_user)
+        post :submit_request, { project: source_project, package: package, targetproject: target_project }
+      end
+
+      it { expect(flash[:error]).to eq("Unable to submit: Validation failed: Creator Login #{unconfirmed_user.login} is not an active user") }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: package)) }
+
+      it "does not create a submit request and shows an error message" do
+        expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: package.name)).not_to exist
+      end
+    end
+
+    context "unchanged sources" do
+      before do
+        login(user)
+        post :submit_request, { project: source_project, package: package, targetproject: source_project }
+      end
+
+      it { expect(flash[:error]).to eq("Unable to submit, sources are unchanged") }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: package)) }
+
+      it "doesn't create a submit request" do
+        expect(BsRequestActionSubmit.where(target_project: source_project.name, target_package: package.name)).not_to exist
+      end
+    end
+
+    context "invalid request (missing parameters)" do
+      before do
+        login(user)
+        post :submit_request, { project: source_project, package: "", targetproject: source_project }
+      end
+
```

Comment:
Can you add the same expectation as line 76, or this creates the submit request?